### PR TITLE
Run rubocop -A

### PR DESCRIPTION
**Pull request by agrare**

### CODE REVIEW

File: inventory_refresh.gemspec [hunk 1]
```diff
@@ -33,4 +32,5 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "rake", "~> 10.0"
   spec.add_development_dependency "rspec", "~> 3.0"
   spec.add_development_dependency "simplecov"
+  spec.metadata['rubygems_mfa_required'] = 'true'
```
Comment (agrare) [inline inventory_refresh.gemspec:35]: Haven't seen this one before.... I'm all for 2fa for rubygems not sure we want to require it yet though @Fryguy ?

File: Rakefile [hunk 2]
```diff
@@ -2,25 +2,25 @@ require "bundler/gem_tasks"
 require "rspec/core/rake_task"
 
 namespace :spec do
-  task :db_drop do
+  task :db_drop => :environment do
```
Comment (Fryguy) [inline Rakefile:5]: Is this right?  do you have an environment task?

File: lib/inventory_refresh/graph/topological_sort.rb [hunk 3]
```diff
@@ -45,14 +45,14 @@ def topological_sort
           max_depth -= 1
           if max_depth <= 0
             message = "Max depth reached while doing topological sort, your graph probably contains a cycle"
-            #logger.error("#{message}:\n#{graph.to_graphviz}")
+            # logger.error("#{message}:\n#{graph.to_graphviz}")
```
Comment (agrare) [inline lib/inventory_refresh/graph/topological_sort.rb:48]: Not sure why this was left commented out in the first place and not simply deleted

File: inventory_refresh.gemspec (hunk 1 above)
Comment (Fryguy) [inline inventory_refresh.gemspec:35]: Not sure why it enforces this, but we probably want it up above on like line 22

File: lib/inventory_refresh/inventory_collection.rb [hunk 4]
```diff
@@ -92,14 +92,8 @@ class InventoryCollection
...
-                :inventory_object_attributes, :name, :saver_strategy, :targeted_scope, :default_values,
-                :targeted_arel, :targeted, :manager_ref_allowed_nil, :use_ar_object,
-                :created_records, :updated_records, :deleted_records, :retention_strategy,
-                :custom_reconnect_block, :batch_extra_attributes, :references_storage, :unconnected_edges,
-                :assert_graph_integrity
+    attr_accessor :attributes_blacklist, :attributes_whitelist
+    attr_reader :model_class, :strategy, :custom_save_block, :parent, :internal_attributes, :delete_method, :dependency_attributes, :manager_ref, :create_only, :association, :complete, :update_only, :transitive_dependency_attributes, :check_changed, :arel, :inventory_object_attributes, :name, :saver_strategy, :targeted_scope, :default_values, :targeted_arel, :targeted, :manager_ref_allowed_nil, :use_ar_object, :created_records, :updated_records, :deleted_records, :retention_strategy, :custom_reconnect_block, :batch_extra_attributes, :references_storage, :unconnected_edges, :assert_graph_integrity
```
Comment (agrare) [inline lib/inventory_refresh/inventory_collection.rb:96]: ETOOMANYATTRIBUTES :laughing:

File: lib/inventory_refresh/application_record_iterator.rb [hunk 5]
```diff
@@ -26,11 +26,9 @@ def initialize(inventory_collection: nil, manager_uuids_set: nil, iterator: nil,
...
-    def find_in_batches(batch_size: 1000)
+    def find_in_batches(batch_size: 1000, &block)
       if iterator
-        iterator.call do |batch|
-          yield(batch)
-        end
+        iterator.call(&block)
       elsif query
         manager_uuids_set.each_slice(batch_size) do |batch|
           yield(query.where(inventory_collection.targeted_selection_for(batch)))
```
Comment (Fryguy) [inline lib/inventory_refresh/application_record_iterator.rb:34]: Strange that it didn't use block.call here.

File: lib/inventory_refresh/inventory_collection.rb (hunk 4 above)
Comment (Fryguy) [inline lib/inventory_refresh/inventory_collection.rb:96]: 🤣

File: lib/inventory_refresh/inventory_collection.rb [hunk 6]
```diff
@@ -92,14 +92,8 @@ class InventoryCollection
...
 
-    attr_reader :model_class, :strategy, :attributes_blacklist, :attributes_whitelist, :custom_save_block, :parent,
-                :internal_attributes, :delete_method, :dependency_attributes, :manager_ref, :create_only,
-                :association, :complete, :update_only, :transitive_dependency_attributes, :check_changed, :arel,
-                :inventory_object_attributes, :name, :saver_strategy, :targeted_scope, :default_values,
-                :targeted_arel, :targeted, :manager_ref_allowed_nil, :use_ar_object,
-                :created_records, :updated_records, :deleted_records, :retention_strategy,
-                :custom_reconnect_block, :batch_extra_attributes, :references_storage, :unconnected_edges,
-                :assert_graph_integrity
+    attr_accessor :attributes_blacklist, :attributes_whitelist
```
Comment (Fryguy) [inline lib/inventory_refresh/inventory_collection.rb:95]: Why did it split these out?

File: Rakefile (hunk 2 above)
Comment (agrare) [reply Rakefile:5]: `Don't know how to build task 'environment' (see --tasks)`

File: lib/inventory_refresh/inventory_collection.rb [hunk 7]
```diff
@@ -232,13 +226,13 @@ def unique_index_keys
 
     # @return [Array<ActiveRecord::ConnectionAdapters::IndexDefinition>] array of all unique indexes known to model
     def unique_indexes
-      @unique_indexes_cache if @unique_indexes_cache
+      @unique_indexes_cache
```
Comment (Fryguy) [inline lib/inventory_refresh/inventory_collection.rb:229]: Was this a bug?  Right now it's a no-op

File: Rakefile (hunk 2 above)
Comment (agrare) [reply Rakefile:5]: Seems like an overzealous rubocop "fix"

File: lib/inventory_refresh/inventory_collection.rb [hunk 8]
```diff
@@ -251,7 +245,7 @@ def unique_indexes
     # @return [ActiveRecord::ConnectionAdapters::IndexDefinition] unique index fitting the keys
     def unique_index_for(keys)
       @unique_index_for_keys_cache ||= {}
-      @unique_index_for_keys_cache[keys] if @unique_index_for_keys_cache[keys]
+      @unique_index_for_keys_cache[keys]
```
Comment (Fryguy) [inline lib/inventory_refresh/inventory_collection.rb:248]: Similarly here...not sure what this is doing.

File: lib/inventory_refresh/inventory_collection.rb [hunk 9]
```diff
@@ -597,8 +593,6 @@ def new_inventory_object(hash)
       inventory_object_class.new(self, hash)
     end
 
-    attr_writer :attributes_blacklist, :attributes_whitelist
-
```
Comment (Fryguy) [inline lib/inventory_refresh/inventory_collection.rb:601]: Oh here's the write bit...how strange.

File: lib/inventory_refresh/inventory_collection.rb (hunk 7 above)
Comment (agrare) [reply lib/inventory_refresh/inventory_collection.rb:229]: yeah I think so, I'm guessing this was supposed to be `return @unique_indexes_cache if @unique_indexes_cache`

File: lib/inventory_refresh/inventory_collection/builder.rb [hunk 10]
```diff
@@ -4,17 +4,17 @@ class Builder
...
+           attributes_blacklist attributes_whitelist batch_extra_attributes
+           complete create_only custom_save_block
+           custom_reconnect_block default_values delete_method
+           dependency_attributes check_changed inventory_object_attributes
+           manager_ref manager_ref_allowed_nil manager_uuids
+           model_class name parent
+           parent_inventory_collections retention_strategy strategy
+           saver_strategy secondary_refs targeted
+           targeted_arel update_only use_ar_object
+           assert_graph_integrity].to_set
```
Comment (Fryguy) [inline lib/inventory_refresh/inventory_collection/builder.rb:17]: Not sure about this one - thinking one-per-line might be easier to read, cause this is tough to read.

File: lib/inventory_refresh/inventory_collection/builder.rb [hunk 11]
```diff
@@ -76,8 +76,8 @@ def construct_data
 
         send(@name.to_sym) if @name.respond_to?(:to_sym) && respond_to?(@name.to_sym)
 
-        if @properties[:model_class].nil?
-          add_properties(:model_class => auto_model_class) unless @options[:without_model_class]
+        if @properties[:model_class].nil? && !(@options[:without_model_class])
```
Comment (Fryguy) [inline lib/inventory_refresh/inventory_collection/builder.rb:79]:
```suggestion
        if @properties[:model_class].nil? && !@options[:without_model_class]
```

**Comment by Fryguy** [comment]
OK, ETOOMANY weird things...I figure you can eyeball these and do a first pass since we are seeing the same issues.  Then after that I'll do a pass and see if there's anything else.

File: .rubocop_local.yml [hunk 12]
```diff
@@ -1,2 +1,5 @@
 # GlobalVars:
 #   AllowedVariables:
+AllCops:
+  Exclude:
```
Comment (Fryguy) [inline .rubocop_local.yml:4]: I think we should exclude all Rails conventions here.  That's what's causing the things like the rake environment stuff.

File: lib/inventory_refresh/inventory_collection/index/proxy.rb [hunk 13]
```diff
@@ -224,7 +228,7 @@ def assert_index(manager_uuid, ref)
             assert_relation_keys({named_ref(ref).first => manager_uuid}, ref)
           end
         rescue => e
-          #_log.error("Error when asserting index: #{manager_uuid}, with ref: #{ref} of: #{inventory_collection}")
+          # _log.error("Error when asserting index: #{manager_uuid}, with ref: #{ref} of: #{inventory_collection}")
```
Comment (agrare) [inline lib/inventory_refresh/inventory_collection/index/proxy.rb:231]: This looks like it should be logged, going to use the InventoryRefresh.logger

File: .rubocop_local.yml [hunk 14]
```diff
@@ -1,2 +1,5 @@
-# GlobalVars:
-#   AllowedVariables:
+AllCops:
+  Exclude:
+  - spec/schema.rb
+Rails:
+  Enabled: false
```
Comment (agrare) [inline .rubocop_local.yml:5]: Good call this helped a lot, re-ran rubocop -A to not pick these up

File: lib/inventory_refresh/inventory_collection.rb (hunk 9 above)
Comment (agrare) [reply lib/inventory_refresh/inventory_collection.rb:601]: Yeah unclear why this was way down away from the others, but not private

File: lib/inventory_refresh/inventory_collection/helpers/questions_helper.rb [hunk 15]
```diff
@@ -52,7 +52,7 @@ def saveable?
 
         # @return [Boolean] true if we are using a saver strategy that allows saving in parallel processes
         def parallel_safe?
-          @parallel_safe_cache ||= %i(concurrent_safe concurrent_safe_batch).include?(saver_strategy)
```
Comment (agrare) [inline lib/inventory_refresh/inventory_collection/helpers/questions_helper.rb:55]: Double checked that no one is referencing the ivar `@parallel_safe_cache` directly and no one is

File: lib/inventory_refresh/inventory_collection/index/proxy.rb [hunk 16]
```diff
@@ -224,7 +228,7 @@ def assert_index(manager_uuid, ref)
             assert_relation_keys({named_ref(ref).first => manager_uuid}, ref)
           end
         rescue => e
-          #_log.error("Error when asserting index: #{manager_uuid}, with ref: #{ref} of: #{inventory_collection}")
+          logger.error("Error when asserting index: #{manager_uuid}, with ref: #{ref} of: #{inventory_collection}")
```
Comment (agrare) [inline lib/inventory_refresh/inventory_collection/index/proxy.rb:231]: This class has `include InventoryRefresh::Logging` already so good to use the `logger` here

File: lib/inventory_refresh/graph.rb [hunk 17]
```diff
@@ -37,7 +37,7 @@ def to_graphviz(layers: nil)
         s << "  #{node_names[from]} -> #{node_names[to]};"
       end
       s << "}"
-      s.join("\n") + "\n"
+      "#{s.join("\n")}\n"
```
Comment (kbrock) [inline lib/inventory_refresh/graph.rb:40]: I totally saw this going with using "<<" instead of string interpolation

**Review by kbrock — APPROVED** [review]
From what I could tell, there didn't seem to be any "lets not do this" changes
Good call on the `Rails: Enable: false`

I say YOLO

**Comment by kbrock** [comment]
are we thinking the failure for one and not the other means this is sporadic?
Or are we thinking this broke something?
(maybe a bug in rubocop)

**Comment by agrare** [comment]
Yeah I think rubocop actually broke some code, considering just dropping this or just having it fix the indentation and `%[]` vs `%()` issues

File: lib/inventory_refresh/graph.rb (hunk 17 above)
Comment (Fryguy) [reply lib/inventory_refresh/graph.rb:40]: Same

Comment (Fryguy) [reply lib/inventory_refresh/graph.rb:40]: I bet if `<<` were used, then it wouldn't have tried to autoconvert, much line the previous line.  @agrare what do you like better?

File: lib/inventory_refresh/inventory_collection.rb [hunk 18]
```diff
@@ -232,13 +227,13 @@ def unique_index_keys
 
     # @return [Array<ActiveRecord::ConnectionAdapters::IndexDefinition>] array of all unique indexes known to model
     def unique_indexes
-      @unique_indexes_cache if @unique_indexes_cache
+      return @unique_indexes_cache if @unique_indexes_cache
```
Comment (Fryguy) [inline lib/inventory_refresh/inventory_collection.rb:230]: I'm wonder if this should be split out into its own PR (along with the other cache thing), because it is an actual change in how the code works.  I'm concerned we might need some extra specs around it, but don't want to muddy up a simpler rubocop PR.

File: lib/inventory_refresh/inventory_collection/builder.rb [hunk 19]
```diff
@@ -4,17 +4,38 @@ class Builder
...
+          parent_inventory_collections
+          retention_strategy
+          strategy
+          saver_strategy
+          secondary_refs
+          targeted
+          targeted_arel
+          update_only
+          use_ar_object
+          assert_graph_integrity
```
Comment (Fryguy) [inline lib/inventory_refresh/inventory_collection/builder.rb:37]: sort for readability / diffs?

File: lib/inventory_refresh/save_collection/saver/base.rb [hunk 20]
```diff
@@ -234,12 +232,12 @@ def assert_distinct_relation(primary_key_value)
           # Change the InventoryCollection's :association or :arel parameter to return distinct results. The :through
           # relations can return the same record multiple times. We don't want to do SELECT DISTINCT by default, since
           # it can be very slow.
-          unless inventory_collection.assert_graph_integrity
+          if inventory_collection.assert_graph_integrity
```
Comment (Fryguy) [inline lib/inventory_refresh/save_collection/saver/base.rb:235]: What a strangely named method...it's an assert_graph_integrity would imply it would return true if the graph has good integrity, but instead it returns false. 😖

File: lib/inventory_refresh/save_collection/saver/sql_helper.rb [hunk 21]
```diff
@@ -157,14 +157,14 @@ def load_scope(all_manager_uuids_scope)
         all_manager_uuids_scope.map do |cond|
           assert_scope!(scope_keys, cond)
 
-          cond.map do |key, value|
+          cond.to_h do |key, value|
```
Comment (Fryguy) [inline lib/inventory_refresh/save_collection/saver/sql_helper.rb:160]: WAT...this is _awesome_....I had no idea to_h could take a block!

File: lib/inventory_refresh/save_collection/saver/sql_helper_upsert.rb [hunk 22]
```diff
@@ -107,10 +108,12 @@ def insert_query_on_conflict_update(all_attribute_keys, mode, ignore_cols, colum
       end
 
       def insert_query_on_conflict_update_mode(mode, version_attribute, column_name)
-        if mode == :full
+        case mode
```
Comment (Fryguy) [inline lib/inventory_refresh/save_collection/saver/sql_helper_upsert.rb:111]: Same here...Minor, but we really should have an else clause with a `raise` (not for this PR)

File: lib/inventory_refresh/save_collection/sweeper.rb [hunk 23]
```diff
@@ -28,8 +28,8 @@ def sweep_possible?(inventory_collection, refresh_state)
       def in_scope?(inventory_collection, sweep_scope)
         return true unless sweep_scope
 
-        if sweep_scope.kind_of?(Array)
-          return true if sweep_scope.include?(inventory_collection&.name&.to_s)
+        if sweep_scope.kind_of?(Array) && sweep_scope.include?(inventory_collection&.name&.to_s)
+          return true
         end
 
         false
```
Comment (Fryguy) [inline lib/inventory_refresh/save_collection/sweeper.rb:35]:
```suggestion
        sweep_scope.kind_of?(Array) && sweep_scope.include?(inventory_collection&.name&.to_s)
```

**Comment by kbrock** [comment]
Yea. these should be no brainers, so anything that is a question mark should probably just remove the option that did those changes and proceed from there? I like Adam's suggestion of just doing the indentation and `() -> []`.

File: lib/inventory_refresh/application_record_iterator.rb (hunk 5 above)
Comment (kbrock) [reply lib/inventory_refresh/application_record_iterator.rb:34]: also, `yield` is faster than `call`. So using a yield would probably be preferred

File: lib/inventory_refresh/inventory_collection.rb (hunk 4 above)
Comment (kbrock) [reply lib/inventory_refresh/inventory_collection.rb:97]: amusing that they felt scrolling off the side of the screen was better than making it so you can read it.

Almost feels like a passive aggressive way of them asking you to reduce the number of attributes.

File: lib/inventory_refresh/inventory_collection.rb [hunk 24]
```diff
@@ -232,16 +227,16 @@ def unique_index_keys
 
     # @return [Array<ActiveRecord::ConnectionAdapters::IndexDefinition>] array of all unique indexes known to model
     def unique_indexes
-      @unique_indexes_cache if @unique_indexes_cache
+      return @unique_indexes if @unique_indexes
```
Comment (kbrock) [inline lib/inventory_refresh/inventory_collection.rb:230]: this is a change. I think it is correct, but just saying that this is a change

File: spec/models/archived_mixin.rb [hunk 25]
```diff
@@ -9,12 +9,12 @@ module ArchivedMixin
   def archived?
     !active?
   end
-  alias_method :archived, :archived?
+  alias archived archived?
```
Comment (kbrock) [inline spec/models/archived_mixin.rb:12]: are there subclasses?
Is this the same thing?

File: spec/models/orchestration_stack.rb [hunk 26]
```diff
@@ -4,6 +4,6 @@ class OrchestrationStack < ActiveRecord::Base
   has_ancestry
 
   belongs_to :ext_management_system, :foreign_key => :ems_id
-  has_many   :resources,  :dependent => :destroy, :foreign_key => :stack_id, :class_name => "OrchestrationStackResource"
-  alias_method :orchestration_stack_resources,  :resources
+  has_many   :resources, :dependent => :destroy, :foreign_key => :stack_id, :class_name => "OrchestrationStackResource"
+  alias orchestration_stack_resources resources
```
Comment (kbrock) [inline spec/models/orchestration_stack.rb:8]: ditto on this fix.

OrchestrationStack has always been confusing for me and I thought there may be some subclasses with it

File: spec/save_inventory/spec_helper.rb [hunk 27]
```diff
@@ -1,7 +1,7 @@
 module SpecHelper
   def assert_all_records_match_hashes(model_classes, *expected_match)
     # Helper for matching attributes of the model's records to an Array of hashes
-    model_classes = model_classes.kind_of?(Array) ? model_classes : [model_classes]
+    model_classes = [model_classes] unless model_classes.kind_of?(Array)
```
Comment (kbrock) [inline spec/save_inventory/spec_helper.rb:4]: Think this is the same thing:

```suggestion
    model_classes = Array.wrap(model_classes)
```

**Review by kbrock — COMMENTED** [review]
Just a few questions on `alias_method`
`alias` is better for this purpose, but I thought there was a nuance for child classes and the two. (or maybe that is the difference between `alias_method :a, :b` and `def a ; b ; end`)

File: lib/inventory_refresh/inventory_collection/helpers/questions_helper.rb [hunk 28]
```diff
@@ -52,13 +52,13 @@ def saveable?
...
+          @parallel_safe ||= %i[concurrent_safe concurrent_safe_batch].include?(saver_strategy)
```
[the remark this answers is not in the document]
Comment (agrare) [reply lib/inventory_refresh/inventory_collection/helpers/questions_helper.rb:55]: Rubocop was warning about the ivar name not matching the method name which is why this is changed, agree on the `.nil?` though will update

File: lib/inventory_refresh/inventory_collection.rb [hunk 29]
```diff
@@ -232,16 +227,16 @@ def unique_index_keys
 
     # @return [Array<ActiveRecord::ConnectionAdapters::IndexDefinition>] array of all unique indexes known to model
     def unique_indexes
-      @unique_indexes_cache if @unique_indexes_cache
+      @unique_indexes if @unique_indexes
```
Comment (Fryguy) [inline lib/inventory_refresh/inventory_collection.rb:230]: @agrare Are you fixing this in a follow up?

Comment (agrare) [reply lib/inventory_refresh/inventory_collection.rb:230]: Yes I split this out into its own PR since it wasn't related to the rubocop changes specifically
